plugins/flatpak/gs-flatpak.c

```diff
@@ -1301,6 +1320,77 @@ gs_flatpak_app_install_source (GsFlatpak *self, GsApp *app,
 	return TRUE;
 }
 
+static GsApp *
+get_main_app_of_related (GsFlatpak *self,
+			 GsApp *related_app,
+			 GCancellable *cancellable,
+			 GError **error)
+{
+	g_autofree gchar *main_app_name = g_strdup (gs_flatpak_app_get_ref_name (related_app));
+	g_autoptr(FlatpakInstalledRef) ref = NULL;
+	const gchar *app_name;
+	g_auto(GStrv) app_tokens = NULL;
+
+	app_name = gs_flatpak_app_get_main_app_ref (related_app);
+	if (app_name == NULL) {
+		g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+			     "%s doesn't have a main app set to it.",
+			     gs_app_get_unique_id (related_app));
+		return NULL;
+	}
+
+	app_tokens = g_strsplit (app_name, "/", -1);
+	if (g_strv_length (app_tokens) != 4) {
+		g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA,
+			     "The main app of %s has an invalid name: %s",
+			     gs_app_get_unique_id (related_app), app_name);
+		return NULL;
+	}
+
+	ref = flatpak_installation_get_installed_ref (self->installation,
+						      FLATPAK_REF_KIND_APP,
+						      app_tokens[1],
+						      app_tokens[2],
+						      app_tokens[3],
+						      cancellable,
+						      error);
+
+	if (ref == NULL)
+		return NULL;
+
+	return gs_flatpak_create_installed (self, ref, error);
+}
+
+static GsApp *
+get_real_app_for_update (GsFlatpak *self, GsApp *app, GCancellable *cancellable,
+			GError **error)
+{
+	GsApp *main_app = NULL;
+	g_autoptr(GError) error_local = NULL;
```

Comment:
It looks like you're using this `error_local` so that you can simply report with a warning when `error != NOT_FOUND`, but then that means you don't need to pass an `GError **error` parameter to this `get_real_app_for_update()` function, as it would be guaranteed to always return a valid `GsApp*`, and never `NULL`.